### PR TITLE
feat: Add record navigation

### PR DESCRIPTION
**Pull request by EdwinBetanc0urt**
#### Screenshot or Gif
record-navigation

#### Other relevant information
* Your OS: Linux Mint 19.1 Cinnamon x64. 
* Browser: Mozilla Firefox 88.0.1.
* Node.js version: 14.17.0.
* NPM version: 7.19.0.
* adempiere-vue version: 4.3.1.

### CODE REVIEW

File: src/components/ADempiere/RecordNavigation/index.vue [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <https:www.gnu.org/licenses/>.
+-->
+
+<template>
+  <el-container class="record-navigation">
+
+    <!-- advanced search -->
+    <!--
```
Comment (yamelsenih) [inline src/components/ADempiere/RecordNavigation/index.vue]: Remove this if is unused

**Review by yamelsenih — CHANGES_REQUESTED** [review]

File: src/components/ADempiere/RecordNavigation/index.vue (hunk 1 above)
Comment (EdwinBetanc0urt) [reply src/components/ADempiere/RecordNavigation/index.vue]: unused code has been removed

**Comment by EdwinBetanc0urt** [comment]
Records navigation with drawer

https://user-images.githubusercontent.com/20288327/124390337-18ea4980-dcb9-11eb-9429-78803268e35f.mp4

File: src/store/modules/ADempiere/dataManager.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        // update records
+        state.containerData.splice(index, 0, payload)
+      } else {
+        state.containerData.push(payload)
+      }
+    }
+  },
+
+  actions: {
+    getListEntities({
```
Comment (yamelsenih) [inline src/store/modules/ADempiere/dataManager.js]: Change to **getEntitiesList** instead

File: src/store/modules/ADempiere/dataManager.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+      commit
+    }, {
+      parentUuid,
+      containerUuid,
+      tableName,
+      query,
```
Comment (yamelsenih) [inline src/store/modules/ADempiere/dataManager.js:33]: remove this line

File: src/store/modules/ADempiere/dataManager.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+      whereClause,
+      pageToken,
+      pageSize
+    }) {
+      // TODO: Parsed query and where clause
+      return new Promise((resolve) => {
+        requestListEntities({
+          tableName,
+          query,
```
Comment (yamelsenih) [inline src/store/modules/ADempiere/dataManager.js:42]: Remove this line

File: src/components/ADempiere/RecordNavigation/index.vue [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+
+    const actionAdvancedQuery = () => {
+      const activeNames = []
+      if (!activeName.value.length) {
+        activeNames.push('PanelAdvancedQuery')
+      }
+      activeName.value = activeNames
+    }
+
+    const recordNavigationManager = computed(() => {
```
Comment (yamelsenih) [inline src/components/ADempiere/RecordNavigation/index.vue]: What is it?

File: src/components/ADempiere/RecordNavigation/index.vue [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+-->
+
+<template>
+  <el-container class="record-navigation">
+
+    <!-- records in table -->
+    <default-table
+      :parent-uuid="parentUuid"
+      :container-uuid="containerUuid"
+      :container-manager="recordNavigationManager"
```
Comment (yamelsenih) [inline src/components/ADempiere/RecordNavigation/index.vue:26]: where is psssed the callback?

**Review by yamelsenih — CHANGES_REQUESTED** [review]

File: src/components/ADempiere/DefaultTable/index.vue [hunk 7]
```diff
@@ -163,10 +167,14 @@ export default defineComponent({
       return
     }
 
+    const recordsList = computed(() => {
+      return props.containerManager.data.recordsList || []
```
Comment (yamelsenih) [inline src/components/ADempiere/DefaultTable/index.vue]: Container manager is not a data container, please remove this

**Review by yamelsenih — CHANGES_REQUESTED** [review]

File: src/store/modules/ADempiere/dataManager.js (hunk 4 above)
Comment (EdwinBetanc0urt) [reply src/store/modules/ADempiere/dataManager.js:42]: without the query how will the query work?

File: src/store/modules/ADempiere/dataManager.js (hunk 3 above)
Comment (EdwinBetanc0urt) [reply src/store/modules/ADempiere/dataManager.js:33]: without the query how will the query work?

**Pull request merged**